### PR TITLE
Revert to python implementation if there is an issue loading the dll

### PR DESCRIPTION
**Pull request by wesen**

### CODE REVIEW

File: uavcan/dsdl/signature.py [hunk 1]
```diff
@@ -18,51 +18,84 @@
...
-            self._crc ^= (b << 56) & Signature.MASK64
-            for _ in range(8):
-                if self._crc & (1 << 63):
-                    self._crc = ((self._crc << 1) & Signature.MASK64) ^ Signature.POLY
-                else:
-                    self._crc <<= 1
+        if Signature.crc64lib is not None:
+            self._crc = Signature.crc64lib.crc64(data_bytes, num_bytes, self._crc)
+        else:
+            if not Signature.WARNED_MISSING_DLL:
```
Comment (daniel-formlabs) [inline uavcan/dsdl/signature.py]: Why not just warn while loading the DLL?

Comment (wesen) [reply uavcan/dsdl/signature.py]: morning brain

File: uavcan/dsdl/signature.py [hunk 2]
```diff
@@ -18,51 +18,84 @@
...
     MASK64 = 0xFFFFFFFFFFFFFFFF
     POLY = 0x42F0E1EBA9EA3693
 
+    crc64lib = None
+
+    WARNED_MISSING_DLL = False
+
+    dll_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                            '../fastcrc64.{}.so'.format(sysconfig.get_config_vars('SOABI')[0]))
+    if os.path.exists(dll_path):
```
Comment (robweiss) [inline uavcan/dsdl/signature.py]: Can we log a message when the path does not exist?

**Review by daniel-formlabs — APPROVED** [review]
LGTM

**Pull request merged**